### ISSUE TITLE
Brush transform doesn't do what it is supposed to

### ISSUE THREAD

**Issue by BAndysc** [AvaloniaUI/Avalonia#15114]
### Describe the bug

https://github.com/AvaloniaUI/Avalonia/pull/14978 changed what the transform does on brushes and now it is bugged.

After:
![now](https://github.com/AvaloniaUI/Avalonia/assets/5689666/7319c67a-a626-40ab-b246-845bd0f99ecc)

### To Reproduce

1. Clone https://github.com/BAndysc/avalonia-bugs-repro
2. Checkout bug_brush_transform
3. Run and press the button
4. See the background disappears

### Expected behavior

I don't know what the transform does now, but previously it would just translate the grid.

Expected behaviour (before the faulty PR):
![before](https://github.com/AvaloniaUI/Avalonia/assets/5689666/2c4558a2-659b-424c-8b10-3d574adcd731)

### Avalonia version

11.2.999-cibuild0046229-beta

**Comment by jp2masa** [comment]
Was fixed in #15307, corresponding to version `11.2.999-cibuild0046385-alpha`.

**Comment by BAndysc** [comment]
Not really, this is something else, I should have said I have tested also against the lastest master

**Comment by maxkatz6** [comment]
cc @Gillibald

**Comment by maxkatz6** [comment]
Was it fixed in the master? After https://github.com/AvaloniaUI/Avalonia/pull/15289 or https://github.com/AvaloniaUI/Avalonia/pull/15157

**Comment by BAndysc** [comment]
Yes, it works now on master
